packages/cli/src/util/input/edit-project-settings.ts

```diff
@@ -2,45 +2,109 @@ import inquirer from 'inquirer';
+  PartialProjectSettings,
+  'framework'
+>;
+type PartialProjectSettingsFrameworkOmittedKeys =
+  keyof Required<PartialProjectSettingsFrameworkOmitted>;
+const fields: {
+  readonly [k in PartialProjectSettingsFrameworkOmittedKeys]: string;
+} = {
+  buildCommand: 'Build Command',
+  devCommand: 'Development Command',
+  ignoreCommand: 'Ignore Command',
+  installCommand: 'Install Command',
+  outputDirectory: 'Output Directory',
+};
 
 export default async function editProjectSettings(
   output: Output,
-  projectSettings: PartialProjectSettings | null,
+  projectSettings: PartialProjectSettingsFrameworkOmitted | null,
   framework: Framework | null,
-  autoConfirm?: boolean
+  autoConfirm: boolean,
+  localConfigurationOverrides: PartialProjectSettings | null
 ): Promise<ProjectSettings> {
-  // create new settings object, missing values will be filled with `null`
+  // Create initial settings object defaulting everything to `null` and assigning what may exist in `projectSettings`
   const settings: ProjectSettings = Object.assign(
-    { framework: null },
+    {
+      buildCommand: null,
+      devCommand: null,
+      framework: null,
+      ignoreCommand: null,
+      installCommand: null,
+      outputDirectory: null,
+    },
     projectSettings
   );
 
-  for (let field of fields) {
-    settings[field.value] =
-      (projectSettings && projectSettings[field.value]) || null;
+  // Start UX by displaying overrides. They will be referenced throughout remainder of CLI.
+  if (localConfigurationOverrides) {
+    // Apply local overrides (from `vercel.json`)
+    Object.assign(settings, localConfigurationOverrides);
+
+    output.print(
+      `Local configuration overrides detected. Overwritten project settings:\n`
+    );
+
+    // Print provided overrides including framework
+    for (const [settingValue, settingDisplayName] of Object.entries(
+      fields
+    ).concat(['framework', 'Framework'])) {
+      const override =
+        localConfigurationOverrides?.[
+          settingValue as PartialProjectSettingsKeys
+        ];
+      if (override) {
+        output.print(
+          `${chalk.dim(
+            `- ${chalk.bold(`${settingDisplayName}:`)} ${override}`
+          )}\n`
+        );
+      }
+    }
   }
 
-  // skip editing project settings if no framework is detected
+  // Apply local framework override
+  if (localConfigurationOverrides?.framework) {
+    // Unfortunately the frameworks map is a readonly constant and does not actually return `Framework` type objects
+    framework = (frameworks.find(
+      _framework => _framework.slug === localConfigurationOverrides.framework
+    ) ?? null) as Framework | null;
+
+    if (framework) {
+      output.print(
+        `Merging default project settings for framework ${framework.name}. Previously listed overrides are prioritized.\n`
+      );
+    } else {
+      output.print(
+        `Framework slug matching ${localConfigurationOverrides.framework} not found.`
```

Comment:
When will a slug exist that doesn't match any framework?
Should this be an error instead?

Reply:
Good point - the api already validates against the list of known slugs so if the user provides something else the API itself will fail.